packages/core/src/styling/customProperty.ts

```diff
@@ -41,18 +41,72 @@ export function stringifySyntaxNode(node: CssSyntaxNode): string {
   }
 }
 
-/*
-TODO: `initialValue` does not support referencing other properties (e.g. var(--some-other-property))
-To Fix this, we could either statically solve by accessing theme at SSR, but to handle theme-switching at runtime,
-we would instead need to do `:root { --my-property: var(--some-other-property); }` for properties that need to reference other properties.
-*/
 export function renderSyntaxDefinition(
   key: CustomPropertyName,
   { syntax, inherits, initialValue }: CustomPropertyDefinition,
+  defaultTheme: Theme,
 ): string {
+  let value = initialValue
+  if (initialValue?.includes('var(--')) {
+    value = solveVarRecursively(initialValue, defaultTheme)
+  }
+
+  // Fallback in-case of no reference
+  if (!isDefined(value) && syntax.type === 'primitive') {
+    value = FALLBACK_VALUES[syntax.name]
+  }
+
   return `@property ${key} {
   syntax: "${stringifySyntaxNode(syntax)}";
   inherits: ${String(inherits)};
-  initial-value: ${initialValue};
+  initial-value: ${value};
 }`
 }
+
+function solveVarRecursively(initialValue: string, theme: Theme, depth = 0) {
+  // This makes a crazy assumption that no person would create a web of style-variable referencing deeper than 256
+  if (depth > 2 ** 8) {
+    return null
+  }
+
+  const VAR_REGEX = /var\((--[a-zA-Z0-9-_]+)\)/g
+
+  let match
+  while ((match = VAR_REGEX.exec(initialValue))) {
+    const varName = match[1]
+    if (
+      !isDefined(theme.propertyDefinitions?.[varName as CustomPropertyName])
+    ) {
+      return null
+    }
+
+    const value =
+      theme.propertyDefinitions?.[varName as CustomPropertyName].initialValue
```

Comment:
I don't think you need optional chaining here if you you extract `theme.propertyDefinitions?.[varName as CustomPropertyName]` to a const above (before the `if` statement)